components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -230,11 +233,16 @@ public OAuth2IntrospectionResponseDTO buildIntrospectionResponse(OAuth2TokenVali
         // To hold the applicable validators list from all the available validators. This list will be prioritized if we
         // have a token_type_hint.
         List<OAuth2TokenValidator> applicableValidators = new ArrayList<>();
+        boolean isJWTTokenValidation = isJWTTokenValidation(oAuth2Token.getIdentifier());
 
         // If we have a token type hint, we have to prioritize our list.
         if (oAuth2Token.getTokenType() != null) {
             if (tokenValidators.get(oAuth2Token.getTokenType()) != null) {
-                applicableValidators.add(tokenValidators.get(oAuth2Token.getTokenType()));
+                // Ignore bearer token validators if the token is JWT.
+                if (!(isJWTTokenValidation && tokenValidators.get(oAuth2Token.getTokenType()) instanceof
```

Comment:
OAuth2JWTTokenValidator is extending DefaultOAuth2TokenValidator. this can cause a problem, right?

Reply:
Fixed in 1b45f0d9e7b8cd9c2aeedb370194448c09747715